### ISSUE TITLE
Autodetect column names for LiDAR files

### ISSUE THREAD

**Issue by rosepearson** [GeospatialResearch/Digital-Twins#33]
LINZ has changed its conventions for how it stores the tile information. This means the hard coded detection of the 'filename', 'minx/y' and 'max x/y' doesn't work. Instead make the process more flexible.

![image](https://user-images.githubusercontent.com/22883860/181645702-f08f19c5-5bc7-46d2-b563-16975bc21902.png)

See https://github.com/rosepearson/GeoFabrics/blob/main/src/geofabrics/geometry.py for an example. 

![image](https://user-images.githubusercontent.com/22883860/181645878-85936969-ca86-4d14-87df-806db8a91115.png)

**Comment by LukeParky** [comment]
@xandercai This may be the next issue that you work on. Please come and talk to me or @caseyli92 about it before you start.

**Comment by xandercai** [comment]
To be compatible with the new structure of Lidar data (13 columns with Polygon Z), we will save the data with the new structure to another `tileindex` table in the database.

**Comment by rosepearson** [comment]
Thanks for the update @xandercai. Something I wonder is if this data needs to be saved in a database table at all. We will ahve to check with @caseyli92 - is it used for any later digital twin processing? The actual `tileindex`  file is of course used by GeoFabrics to produce the DEM. 

Perhaps something for us to meeting about next Wednesday - even if you've resolved this issue by then @xandercai - as I do wonder if we can drop some of the information we're recording in the database tables now that the digital twin is more advanced and we better understand what is or is not required.

**Comment by LukeParky** [comment]
> Thanks for the update @xandercai. Something I wonder is if this data needs to be saved in a database table at all. We will ahve to check with @caseyli92 - is it used for any later digital twin processing? The actual `tileindex` file is of course used by GeoFabrics to produce the DEM.
> 
> Perhaps something for us to meeting about next Wednesday - even if you've resolved this issue by then @xandercai - as I do wonder if we can drop some of the information we're recording in the database tables now that the digital twin is more advanced and we better understand what is or is not required.

Yes @rosepearson, I would like to join this meeting too. I think it depends on how we are using GeoFabrics and if the raw LiDAR data is being stored. It may be valuable to prevent redownloading raw LiDAR data.

**Comment by xandercai** [comment]
> 

We do need to review the data flow and table to make usage clear and code efficient. At least the current `lidar` table does not look "tidy" to me. It saves the file name in three columns in the `lidar` table.

**Comment by LukeParky** [comment]
> > 
> 
> We do need to review the data flow and table to make usage clear and code efficient. At least the current `lidar` table does not look "tidy" to me. It saves the file name in three columns in the `lidar` table.

Yes @xandercai I agree. We should evaluate which columns are needed, which are not needed at all, and which can be easily generated in code and shouldn't be stored.

**Comment by LukeParky** [comment]
Hi @xandercai, I'm doing some cleanup and wanted to check if this issue is solved by NewZeaLiDAR and the branch can be deleted and issue closed

**Comment by xandercai** [comment]
Hi @LukeParky, thank you for asking. The issue was solved (although the same issue happened in GeoFabrics in the recent version). Feel free to delete the related branch and close the issue.
